extensions/exchanges/gdax/exchange.js

```diff
@@ -69,6 +111,30 @@ module.exports = function container (get, set, clear) {
         // move cursor into the past
         args.after = opts.to
       }
+      // check for locally cached trades from the websocket feed
+      var cache = websocket_cache[opts.product_id]
+      var max_trade_id = cache.trade_ids.reduce(function(a, b) {
+        return Math.max(a, b)
+      }, -1)
+      if (opts.from && max_trade_id >= opts.from) {
+        var fromIndex = cache.trades.findIndex((value)=> {return value.trade_id == opts.from})
+        var newTrades = cache.trades.slice(fromIndex + 1)
+        newTrades = newTrades.map(function (trade) {
+          return {
+            trade_id: trade.trade_id,
+            time: new Date(trade.time).getTime(),
+            size: Number(trade.size),
+            price: Number(trade.price),
+            side: trade.side
+          }
+        })
+        newTrades.reverse()
+        cb(null, newTrades)
+        // trim cache
+        cache.trades = cache.trades.slice(fromIndex)
+        cache.trade_ids = cache.trade_ids.slice(fromIndex)
+        return
+      }
       client.getProductTrades(args, function (err, resp, body) {
```

Comment:
Need to append product_id here change: client.getProductTrades(args, function (err, resp, body) to client.getProductTrades(opts.product_id, args, function (err, resp, body)

Reply:
@krystophv is this something you're still looking at?

Reply:
@DeviaVir it's not an issue any longer since we pinned the dependency instead